test/unit/testStateMachine.js

```diff
@@ -0,0 +1,760 @@
+    beforeEach(() => {
+      const location = new Location(testLocationId)
+      sandbox.stub(redis, 'getLatestState').returns({ state: STATE.STILLNESS_TIMER })
+      sandbox.stub(redis, 'getLatestDoorSensorData').returns({ signal: DOOR_STATE.CLOSED })
+      sandbox.stub(stateMachineHelpers, 'movementAverageOverThreshold').returns(false)
+      const timerStub = sandbox.stub(stateMachineHelpers, 'timerExceeded')
+      timerStub.withArgs(testLocationId, location.durationTimer + location.initialTimer, STATE.INITIAL_TIMER).returns(true)
+      timerStub.withArgs(testLocationId, location.stillnessTimer, STATE.STILLNESS_TIMER).returns(false)
+
+      sandbox.stub(redis, 'addStateMachineData')
+    })
+
+    afterEach(() => {
+      sandbox.restore()
+    })
+
+    it('should change state to IDLE', async () => {
+      const handleAlert = sinon.stub()
+      await StateMachine.getNextState(new Location(testLocationId), handleAlert)
+      expect(redis.addStateMachineData).to.be.calledWith(STATE.IDLE)
+    })
+
+    it('should issue a duration alert', async () => {
+      const handleAlert = sinon.stub()
+      await StateMachine.getNextState(new Location(testLocationId), handleAlert)
+      expect(handleAlert).to.be.calledWith(new Location(testLocationId), ALERT_REASON.DURATION)
+    })
+  })
+
+  describe('and the door is closed, and movement is under the threshold and only the stillness timer has elapsed', async () => {
+    beforeEach(() => {
+      const location = new Location(testLocationId)
+      sandbox.stub(redis, 'getLatestState').returns({ state: STATE.STILLNESS_TIMER })
+      sandbox.stub(redis, 'getLatestDoorSensorData').returns({ signal: DOOR_STATE.CLOSED })
+      sandbox.stub(stateMachineHelpers, 'movementAverageOverThreshold').returns(false)
+      const timerStub = sandbox.stub(stateMachineHelpers, 'timerExceeded')
+      timerStub.withArgs(testLocationId, +location.durationTimer + +location.initialTimer, STATE.INITIAL_TIMER).returns(false)
+      timerStub.withArgs(testLocationId, +location.stillnessTimer, STATE.STILLNESS_TIMER).returns(true)
+
+      sandbox.stub(redis, 'addStateMachineData')
+    })
+
+    afterEach(() => {
+      sandbox.restore()
+    })
+
+    it('should change state to IDLE', async () => {
+      const handleAlert = sinon.stub()
+      await StateMachine.getNextState(new Location(testLocationId), handleAlert)
+      expect(redis.addStateMachineData).to.be.calledWith(STATE.IDLE)
+    })
+
+    it('should issue a stillness alert', async () => {
+      const handleAlert = sinon.stub()
+      await StateMachine.getNextState(new Location(testLocationId), handleAlert)
+      expect(handleAlert).to.be.calledWith(new Location(testLocationId), ALERT_REASON.STILLNESS)
+    })
+  })
+
+  describe('and the door is closed, and movement is under the threshold and only the stillness timer has elapsed', async () => {
```

Comment:
I don't know if that makes sense as a test - because if the duration timer has elapsed, then the stillness timer would have already elapsed and it would have gone back to idle.

Reply:
nothing says the stillness timer has to be shorter, but I think the same is true of the other case, where the duration timer has gone off, but not stillness - and in that case we also go back to idle